### PR TITLE
fix(files_sharing): Adjust design of account filter for file list

### LINKED ISSUE #46771: Redesign broken on mobile
Just tested on daily.ltd2.nextcloud.com:

Screenshot_20240726_101124_Brave
Screenshot_20240726_101143_Brave

FYI @nextcloud/designers

### LINKED ISSUE #46850: [Bug]: Broken file filter UI
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

NcSelect is cut without focus
Bildschirmfoto vom 2024-07-29 15-29-10
NcSelect is intermitted with focus
Bildschirmfoto vom 2024-07-29 15-29-16
Selected filters are cut
image
No alignment
image
Bildschirmfoto vom 2024-07-29 15-36-48

### Steps to reproduce

1. Open /apps/files

### Expected behavior

Good design

### Installation method

None

### Nextcloud Server version

master

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by susnux**
* Resolves https://github.com/nextcloud/server/issues/46771
* Resolves https://github.com/nextcloud/server/issues/46850

## Summary

Fix the design of the file list filter:
image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by AndyScherzinger — APPROVED** [review]
🐘

**Review by Pytal — COMMENTED** [review]
In `updateAvailableAccounts` I think sharees need to be filtered by type as a group share is treated as if it were an account

**Review by jancborchardt — DISMISSED** [review]
Looks great! Can we rename "Accounts" to "People" which sounds friendlier and also fits with the unified search? :)

**Comment by susnux** [comment]
@Pytal and @jancborchardt both of your comments are now implemented, thank you!

image

**Comment by skjnldsv** [comment]
Slight request to the design team.
I think this is a bit technical looking and not super clear we're displaying filters.
Couldn't we add a label before this set of buttons/menus ?

The select menu is a bit more straightforward, as we display a summary of the selection before, and the actions labels speak for themselves with a verb in it
image

Maybe something like that (very rough mùockup)
image

**Comment by skjnldsv** [comment]
> I agree but can we do this in a follow up as this one here is fixing a regression from beta 1?

Absolutely! Sorry, I thought I approved already!! :see_no_evil:

**Review by skjnldsv — APPROVED** [review]

File: apps/files/src/components/FileListFilters.vue [hunk 1]
```diff
@@ -14,7 +14,11 @@
 				<NcChip :aria-label-close="t('files', 'Remove filter')"
 					:icon-svg="chip.icon"
 					:text="chip.text"
-					@close="chip.onclick" />
+					@close="chip.onclick">
+					<template v-if="chip.user" #icon>
+						<NcAvatar :size="24" disable-menu :show-user-status="false" :user="chip.user" />
```
Comment (skjnldsv) [inline apps/files/src/components/FileListFilters.vue:19]:
```suggestion
						<NcAvatar :size="24"
							:disable-menu="true"
							:show-user-status="false"
							:user="chip.user" />
```

Comment (susnux) [reply apps/files/src/components/FileListFilters.vue:19]: I really need more memory :pensive: 
VS Code refuses to properly update the lint results when memory is like ~15/16GB used :sweat_smile:

Comment (skjnldsv) [reply apps/files/src/components/FileListFilters.vue]: :people_hugging:

**Comment by blizzz** [comment]
don't push, i will merge and compile  assets with https://github.com/nextcloud/server/pull/46870

**Comment by jancborchardt** [comment]
Google Drive has the filters listed without a label as well, so it could be fine. Also, the "Filters" text _and_ icon adds even more noise to the file list.

**Comment by jancborchardt** [comment]
For reference this is how Google Drive and Dropbox do the filters, no label: https://github.com/nextcloud/server/issues/43241

**Comment by blizzz** [comment]
@jancborchardt are you intentionally replying to a merged pull request?

**Comment by skjnldsv** [comment]
> For reference this is how Google Drive and Dropbox do the filters, no label: #43241

I guess the left padding is what makes it look weird in my opinion :)
No Google nor Dropbox does it 
image

**Comment by Fenn-CS** [comment]
> I guess the left padding is what makes it look weird in my opinion :)
> No Google nor Dropbox does it

- I agree, the left padding is unnecessary plus the filters keyword should not be added at all users can figure that out.
- The filters should probably come before the recommended section (and those should be filter too)

Screenshot from 2024-07-31 13-14-41


But these should be moved to a follow up ticket/issue since this is already merged and might be spam-ish.

**Comment by susnux** [comment]
> The filters should probably come before the recommended section (and those should be filter too)

This was discussed before and it is the same as with rich workspaces:
The filters should be directly above the file list content. But agree please move discussion to separate ticket :)

**Comment by BloodyIron** [comment]
It's not entirely clear to me, but... is this going to be full-width search presentation? As in, full-width of the page, not just a pull-down menu with limited view space? It looks like yes, but I would like to have this clarified.

Also, is this going to be able to present much larger amounts of results than just 10, and expanding by 10, each time?

Don't play with my heart! I've wanted to go back multiple Search overhauls for years! D:

**Comment by AndyScherzinger** [comment]
@BloodyIron currently it only allows for filtering within the current place you are in within files but yes, actual search is also being looked into but not there yet. it is "in-app" so it is within the files user interface itself. You can already try it out with the v30 beta releases -but again, currently it only implemented filtering since search is a more complex implementation.

**Comment by BloodyIron** [comment]
> @BloodyIron currently it only allows for filtering within the current place you are in within files but yes, actual search is also being looked into but not there yet. it is "in-app" so it is within the files user interface itself. You can already try it out with the v30 beta releases -but again, currently it only implemented filtering since search is a more complex implementation.

In the example of the "Files" app... is it recursive including and below the current scope when triggered? Or is it Files-wide (ALL the files!?) or...?

**Pull request merged**